app/gui2/src/providers/interactionHandler.ts

```diff
@@ -0,0 +1,52 @@
+import { watch, type WatchSource } from 'vue'
+import { createContextStore } from '.'
+
+export { injectFn as injectInteractionHandler, provideFn as provideInteractionHandler }
+const { provideFn, injectFn } = createContextStore(
+  'Interaction handler',
+  () => new InteractionHandler(),
+)
+
+export class InteractionHandler {
+  private currentInteraction: Interaction | undefined = undefined
+
+  /** Automatically activate specified interaction any time a specified condition becomes true. */
+  setWhen(active: WatchSource<boolean>, interaction: Interaction) {
+    watch(active, (active) => {
+      if (active) {
+        this.setCurrent(interaction)
+      } else {
+        this.end(interaction)
+      }
+    })
+  }
+
+  setCurrent(interaction: Interaction | undefined) {
+    if (interaction !== this.currentInteraction) {
+      this.currentInteraction?.cancel?.()
+      this.currentInteraction = interaction
+    }
+  }
+
+  /** Unset the current interaction, if it is the specified instance. */
+  end(interaction: Interaction) {
```

Comment:
Nitpick: I wonder what is the rule of which methods are documented, and which are not. For example, this particular method is documented enough by its code.